src/migration/1686749310179-UpvoteTrigger.ts

```diff
@@ -0,0 +1,42 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class UpvoteTrigger1686749310179 implements MigrationInterface {
+    name = 'UpvoteTrigger1686749310179'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+      queryRunner.query(`
+        CREATE OR REPLACE FUNCTION increment_upvote()
+          RETURNS TRIGGER
+          LANGUAGE PLPGSQL
+          AS
+        $$
+        BEGIN
+          UPDATE post SET upvotes = upvotes + 1 WHERE id = NEW."postId";
+          RETURN NEW;
+        END;
+        $$
+      `)
+      queryRunner.query('CREATE TRIGGER upvote_create_trigger AFTER INSERT ON "upvote" FOR EACH ROW EXECUTE PROCEDURE increment_upvote()')
```

Comment:
CREATE OR REPLACE may be better just in case. Also for the other trigger

Reply:
@idoshamun yeah I tried it but `OR REPLACE` for `CREATE TRIGGER` is only available from version [14](https://www.postgresql.org/docs/14/sql-createtrigger.html), we are on version [11](https://www.postgresql.org/docs/11/sql-createtrigger.html) from what I can see.

Reply:
Oh right! Thanks for letting me know